pkg/process/checks/net.go

```diff
@@ -82,6 +84,8 @@ func (c *ConnectionsCheck) Init(cfg *config.AgentConfig, _ *model.SystemInfo) {
 		log.Infof("no network ID detected: %s", err)
 	}
 	c.networkID = networkID
+	c.dockerFilter = parser.NewDockerProxy()
+	c.serviceExtractor = parser.NewServiceExtractor(cfg.EnableProcessServiceInference)
```

Comment:
Instead of adding the `EnableProcessServiceInference` to the `AgentConfig`, what do you think of reading it directly from global Config? During the config refactor that we did a while ago, we wanted to stop using the `AgentConfig` struct so we should avoid adding new settings to it.

Something similar to what we do here in the Events check:
https://github.com/DataDog/datadog-agent/blob/dafb631bf1634faaca5b445bb4a8118c31f5ff03/pkg/process/events/listener_linux.go#L49